### PR TITLE
[DSS-412] aria-describedby message form utility

### PR DESCRIPTION
**Pull request by QuintonJason**
# Description
Inspired by Kevin's addition in his [Switch PR](https://github.com/Kajabi/sage/pull/55). The PR moves the functions, `assignDescription` and `messageId`, into the `utils` directory and call them in the all form elements that will have helper text

- [x] move functions to utility folder
- [x] update tests accordingly
- [x] make the componentId required for all form elements

Fixes #(issue)
[DSS-412](https://kajabi.atlassian.net/browse/DSS-412)

|  Before  | After  |
|---|---|
|Screen_Shot_2023-06-02_at_2_30_01_PM|Screen_Shot_2023-06-02_at_2_24_33_PM|

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests you've added and run to verify your changes.
Provide instructions so that we can reproduce.
Please also list any relevant details for your test configuration.

Visit the following helper text views for the corresponding form element component and verify, by using the inspector, that the helper text has an `id` that corresponds to the `aria-describedby` for the field:

- [Checkbox](http://localhost:6006/?path=/story/components-checkbox--with-message)
- [Input](http://localhost:6006/?path=/story/components-input--hint)
- [Switch](http://localhost:6006/?path=/story/components-switch--helper-message)
- [Textarea](http://localhost:6006/?path=/story/components-textarea--hint)

## Testing
- [x] unit tests
- [x] e2e tests
- [x] accessibility tests
- [ ] tested manually
- [ ] other:

**Test Configuration**:

- Sage versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

[DSS-412]: https://kajabi.atlassian.net/browse/DSS-412?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pixelflips — COMMENTED** [review]
In the input component, I believe the component-id is missing from the examples. This is causing the id on the message and the aria-describedby to contain undefined in the generated strings.


See screenshot:

Screenshot 2023-06-06 at 11 21 38 AM

**Comment by QuintonJason** [comment]
> In the input component, I believe the component-id is missing from the examples. This is causing the id on the message and the aria-describedby to contain undefined in the generated strings.
> 
> See screenshot:
> 
> Screenshot 2023-06-06 at 11 21 38 AM

Great catch! Added a ticket, https://kajabi.atlassian.net/browse/DSS-417, to address this for this and any case where the developer needs to be alerted

**Review by pixelflips — APPROVED** [review]
Nice work and thanks for the additional clean up as well. LGTM! 👍🏼

File: libs/core/src/utils/form.spec.ts
Comment (ju-Skinner) [inline libs/core/src/utils/form.spec.ts:1]: maybe not now, but soon. I would love to have the test moved to a sub-folder `test`.

File: libs/core/src/utils/form.spec.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { isRequired } from './form';
+
+describe('isRequired', () => {
+  it('returns empty string when no target or component defined', () => {
+    expect(isRequired(undefined, undefined)).toEqual(undefined);
+  });
+});
```
Comment (ju-Skinner) [inline libs/core/src/utils/form.spec.ts]: We should have multiple test for `assignDescription` that covers the various scenarios.

* helperMessage null or undefined
* invalid both true or false
* happy path scenarios

We also need to add test for `messageId`

**Review by ju-Skinner — CHANGES_REQUESTED** [review]
See comments below. TL;DR need test for `form` helper functions

**Review by pixelflips — APPROVED** [review]
LGTM! Rechecked each form input and the new attributes are showing up properly. Great work! 👍🏼

File: libs/core/src/components/pds-input/stories/pds-input.docs.stories.mdx [hunk 2]
```diff
@@ -12,6 +12,7 @@ import * as stories from './pds-input.stories.js';
     disabled: false,
     errorText: null,
     hint: null,
+    invalid: false,
```
Comment (ju-Skinner) [inline libs/core/src/components/pds-input/stories/pds-input.docs.stories.mdx:15]: Do we want an example that shows "invalid"?

**Review by ju-Skinner — APPROVED** [review]

**Pull request merged**